### PR TITLE
Implement new NFT metadata standard in Kitty Items

### PR DESCRIPTION
**Pull request by psiemens**
This PR updates Kitty Items to use the NFT metadata standard. In particular, it implements the [MetadataViews.Display view](https://github.com/onflow/flow-nft/pull/55).

- Updates `KittyItems.cdc` and relevant scripts/transactions to use `MetadataViews`
  - `typeID` is renamed to `kind` to avoid confusing the Kitty Item type (e.g. "skateboard") with the actual Cadence type
  - `rarityID` is renamed to `rarity` and added to the Cadence contract. It was previously only stored off chain
  - Both the above fields are enums which are reflected in the JS client code
  - All Kitty Items images are pinned to IFPS and their CIDs are stored in the contract
- Updates the API to use the new contract and transactions
  - Introduced `Kind` and `Rarity` enums and moved the random minting logic into isolated functions
  - Renamed `item_type` to `item_kind` in database 
- Updates the Next web app to load images from metadata/IPFS rather local filesystem
  - Renamed occurrences of `typeID` to `kind`
  - Renamed occurrences of `rarityID` to `rarity`

### CODE REVIEW

File: cadence/contracts/KittyItems.cdc [hunk 1]
```diff
@@ -23,26 +21,119 @@ pub contract KittyItems: NonFungibleToken {
...
+            case Kind.skateboard:
+                return "Skateboard"
+            case Kind.shades:
+                return "Shades"
+        }
 
+        return ""
+    }
+
+    pub var images: {Kind: {Rarity: String}}
```
Comment (joshuahannan) [inline cadence/contracts/KittyItems.cdc:74]: This dictionary should be `access(self)` or `access(contract)`

File: cadence/contracts/KittyItems.cdc [hunk 2]
```diff
@@ -23,26 +21,119 @@ pub contract KittyItems: NonFungibleToken {
...
+
+    // Rarity -> price rarity mapping
+    pub var itemRarityPriceMap: {Rarity: UFix64}
```
Comment (joshuahannan) [inline cadence/contracts/KittyItems.cdc:77]:
```suggestion
    access(self) var itemRarityPriceMap: {Rarity: UFix64}
```

File: cadence/scripts/kittyItems/get_kitty_item.cdc [hunk 3]
```diff
@@ -1,24 +1,76 @@
 import NonFungibleToken from "../../contracts/NonFungibleToken.cdc"
+import MetadataViews from "../../contracts/MetadataViews.cdc"
 import KittyItems from "../../contracts/KittyItems.cdc"
 
-pub struct AccountItem {
+pub struct KittyItem {
```
Comment (joshuahannan) [inline cadence/scripts/kittyItems/get_kitty_item.cdc:5]: Would it make sense to define this struct in the contract?

File: cadence/scripts/kittyItems/get_kitty_item.cdc [hunk 4]
```diff
@@ -1,24 +1,76 @@
...
-    self.typeID = typeID
     self.resourceID = resourceID
+    self.kind = kind
+    self.rarity = rarity
     self.owner = owner
   }
 }
 
-pub fun main(address: Address, itemID: UInt64): AccountItem? {
+pub fun dwebURL(_ file: MetadataViews.IPFSFile): String {
```
Comment (joshuahannan) [inline cadence/scripts/kittyItems/get_kitty_item.cdc:38]: Would it make sense to include methods like this in the `MetadataViews` contract or maybe in some sort of other utility contract?

File: cadence/scripts/nftStorefront/get_listing_item.cdc [hunk 5]
```diff
@@ -1,36 +1,92 @@
 import NonFungibleToken from "../../contracts/NonFungibleToken.cdc"
+import MetadataViews from "../../contracts/MetadataViews.cdc"
 import NFTStorefront from "../../contracts/NFTStorefront.cdc"
 import KittyItems from "../../contracts/KittyItems.cdc"
 
-pub struct SaleItem {
+pub struct ListingItem {
```
Comment (joshuahannan) [inline cadence/scripts/nftStorefront/get_listing_item.cdc:6]: Again, it feels like this should be something that is defined in the KittyItems contract or storefront or something like that

**Review by joshuahannan — APPROVED** [review]
Looks good! Just a few comments. I only reviewed the cadence parts though, so someone else will have to look at the js stuff.

Looking at this also reminded me that everyone is still hard-coding transactions and scripts into their FCL code and it still hurts my soul. 😅

**Comment by psiemens** [comment]
> Looking at this also reminded me that everyone is still hard-coding transactions and scripts into their FCL code and it still hurts my soul. 😅

@alxflw is looking into this in #192!

File: web/src/components/Minter.jsx [hunk 6]
```diff
@@ -63,19 +63,22 @@ export default function Minter() {
   return (
     <div className="grid grid-cols-1 lg:grid-cols-2">
       {!isLoading && (
-        <ListItemImage typeId={0} rarityId={0} size="lg" grayscale={true} />
+        <ListItemImage
```
Comment (psiemens) [inline web/src/components/Minter.jsx:66]: TODO: We should use a separate image component here since the gray placeholder is not in IPFS.

Comment (10thfloor) [reply web/src/components/Minter.jsx:66]: Created an issue for this.

**Review by 10thfloor — APPROVED** [review]

**Pull request merged**